### ISSUE TITLE
Backpropagation Loss issue?

### ISSUE THREAD

**Issue by andrewzhang214** [EdoardoBotta/RQ-VAE-Recommender#67]
Hello,

I could be crazy here, but I was constructing the computational graph of a forward pass of RQVAE, but it seems like the backpropagation path is cut off when going beyond residuals past the first layer of a codebook?

My understanding of the quantize loss function is that the output of the encoder should try to match the sum of the quantized embeddings.
However, because backpropagation does not go through the second layer, the output of the encoder is really only learning to match the first quantized embedding.

Specifically, shouldn't line 130 in`modules/rqvae.py` be `res = res - emb.detach()`?
Otherwise when we take the gradient over the quantize loss the gradient goes through `emb`, through the straight through estimator and back to the encoding as its negative value. The same gradient goes through `res` as a positive value, when they reach the encoding output it cancels out to 0.

I think if you use the fix above though, the reconstruction loss is now multiplied by a factor of `num_layers`.

Tell me if I'm misunderstanding something here! Thank you!
